### ISSUE TITLE
Create additional packages/workspaces within the monorepo

### ISSUE THREAD

**Issue by benelan** [Esri/calcite-components#7122]
### Summary

The initial monorepo PR is installed! Once we ensure everything is running smoothly, we can start splitting out some of the code in `packages/calcite-components` into separate workspaces. [Turbo has an example project](https://github.com/vercel/turbo/tree/main/examples/design-system) where they have separate workspaces for eslint, tsconfig, and storybook. We could potentially split the demo pages into its own workspace as well.

In addition to creating workspaces from existing code in the monorepo, we can start moving in packages from other repos. Here is a list copied from #6532:

- [x] `@esri/calcite-components-react`
- [ ] `@esri/eslint-plugin-calcite-components`
- [ ] `@esri/calcite-design-tokens`
- [ ] `@esri/calcite-ui-icons` (potentially)
- [ ]  [framework examples](https://github.com/Esri/calcite-components-examples)
- [ ] shared eslint configuration (see turbo's example linked above)
- [ ] tsconfigs (see turbo's example linked above)
- [ ] storybook (see turbo's example linked above)
- [ ] component utils (future)
- [ ] test utils (future)
- [ ] demos (future)

### Desired Outcome

Adding more Calcite packages to the repo and splitting existing functionality into encapsulated workspaces will improve maintainability and discoverability.

**Comment by driskull** [comment]
> @esri/calcite-ui-icons (potentially)

Seems like this would be good to be in the monorepo right? Anything calcite should be in this repo ideally.

**Comment by benelan** [comment]
Yeah I think Franco put the "potentially" there before he talked with @SkyeSeitz and the other maintainers. I just copied it from his list in the issue I linked. But last I heard we are on track to move the icons to the monorepo.

**Comment by SkyeSeitz** [comment]
Thanks, @benelan. Can you remind me of the impact so so I can confirm with the Graphic Illustration team on any updates to their workflow - if any? A url change? Any impacts on maintaining[ this site](https://esri.github.io/calcite-ui-icons/)?

**Comment by benelan** [comment]
Yeah the code would be moved to this repo so they would need to get set up here with permissions etc. The github pages site would need to be moved as well

**Comment by alisonailea** [comment]
I'd like to propose we create a /configs folder at root for all shared configs (typescript, jest, eslint, etc.) and a /tools folder for shared tooling (tokens-transformer)

**Comment by benelan** [comment]
It would just be the shared configs for the design system. Here is an example of a shared eslint config from turbo: 
https://github.com/vercel/turbo/tree/main/examples/design-system/packages/eslint-config-acme

which can be installed and extended in other packages within the repo:
https://github.com/vercel/turbo/blob/main/examples/design-system/packages/acme-core/package.json#L23
https://github.com/vercel/turbo/blob/main/examples/design-system/packages/acme-core/.eslintrc.js#L3

**Comment by benelan** [comment]
> I'd like to propose we create a /configs folder at root for all shared configs (typescript, jest, eslint, etc.) 

I started on this but ran into issues with the typescript parser. Here are some resources for when @alisonailea or I pick it back up:

- https://typescript-eslint.io/linting/typed-linting/monorepos
- https://turbo.build/repo/docs/handbook/linting/typescript
- https://turbo.build/repo/docs/handbook/linting#running-tasks
- I ran into this bug but it shouldn't be an issue once we figure out how to setup the typescript parser correctly: https://github.com/stencil-community/stencil-eslint/issues/15
- everything I did so far is on this branch: `benelan/separate-eslint-tsconfig`

**Comment by geospatialem** [comment]
Since the above has been achieved with the exception of #9257 will be closing out the above epic, where the last remaining issue will be carried out to share configuration across the monorepo. 🤜🏻🤛🏻
